build.sh

```diff
@@ -35,7 +40,7 @@ while [[ "${1:-}" != "" ]]; do
 done
 
 export PATH=$(npm bin):$PATH
-export NODE_OPTIONS="--max-old-space-size=4096 ${NODE_OPTIONS:-}"
+export NODE_OPTIONS="--max-old-space-size=4096 --experimental-worker ${NODE_OPTIONS:-}"
```

Comment:
We probably need to be a bit careful here. 

I believe this option is required only for Node 10. In later versions of node, this has graduated out of experimental. And not supported in earlier versions of node.

What's the experience when this option is set for version > node 10? Fail or ignore?

We could be super precise and say - if node 10 and snippets extraction is set, then use `--experiemental-worker` and if < node10 and snippet extraction is set, fail?

WDYT? too pedantic?

Reply:
*The following is just speculation on how node should behave since I haven't been able to easily find node's policies on experimental features.*

Providing flags that enable experimental features after the feature has stabilized *should not* have any effect on behaviour (confirmed for v10, v12, and v14). Further, the flag *should not* be reused for another purpose. Any other experience would be very frustrating to customers, forcing them to configure different versions in different ways and breaking legacy scripts.

So I don't think it's necessary to be careful with adding this flag. I'm of course happy to do so if you feel otherwise!

---

I also don't think we should fail if the node version is < 10, since rosetta itself still supports single-threaded extraction (though it is quite slow, especially for the monolithic modules). Rosetta already [emits a warning](https://github.com/aws/jsii/blob/d9028c8cf91297a44cd4b1f128f769716bf0e74e/packages/jsii-rosetta/lib/commands/extract.ts#L112) in this case.